### ISSUE TITLE
[Feature]: Use radians for angles in Circle

### ISSUE THREAD

**Issue by lbordowitz** [fabricjs/fabric.js#9654]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have searched and referenced existing issues, feature requests and discussions
- [X] I am filing a **FEATURE** request.

### Description

With a clean break between Fabric 5.0.0 and Fabric 6.0.0, it would be nice to have the angle provided in radians in the Circle class, maybe any other places that are still using degrees.

### Current State

The current code uses degrees instead of radians. That's a bit awkward in some spots.

### Additional Context

The `endAngle` documentation mentions " End angle of the circle probably will change to degrees in next major version ". Was it previously radians, and then changed to degrees? If that's the case, maybe just the documentation needs to change?

**Comment by ShaMan123** [comment]
An effort was made to move all to deg
Why do you need rads?
It is annoying working in rads and needing to convert only for it to be converted back.
I would say that a robust soultion would be to introduce a class
```
class Angle {
  static deg(value: TDegree) 
  static rad(value: TDegree) 

  // impl lazy getters
  get/set deg() 
  get/set rad() 
  
}

```

and have the codebase use this class all over 
with lazy getters it will save compution and make all of us happy

**Comment by asturur** [comment]
Yes it was in radians before and then moved to angle and the documentation should be fixed.
